stps/sig-network/ip-request-stp.md

```diff
@@ -0,0 +1,218 @@
+| **Topology Considerations**      | [ ]  | Evaluated multi-cluster, network topology, and architectural impacts.                                                                                   |          |
+
+
+### **II. Software Test Plan (STP)**
+
+This STP serves as the **overall roadmap for testing**, detailing the scope, approach, resources, and schedule.
+
+#### **1. Scope of Testing**
+
+Briefly describe what will be tested. The scope must **cover functional and non-functional requirements**. Must ensure user stories are included and aligned to downstream user stories from Section I.
+
+**In Scope:**
+- Testing of known networking flows on VMs, where specific IP is requested for primary interface over
+Cluster/UserDefinedNetwork.\
+The uniqueness in this feature is how the IP is assigned - using the new `  network.kubevirt.io/addresses` annotation
+
+**Document Conventions (if applicable):** None
+
+#### **2. Testing Goals**
+
+Define specific, measurable testing objectives for this feature, such as:
+
+- [ ] Goal 1: Verify known networking scenarios are still valid when primary static IP is assigned via the new\
+annotation.
+- [ ] [Goal 2: Consider new cases to be tested, presented by this new feature.
+
+#### **3. Non-Goals (Testing Scope Exclusions)**
+
+Explicitly document what is **out of scope** for testing. **Critical:** All non-goals require explicit stakeholder agreement to prevent "I assumed you were testing that" issues.
+
+**Note:** Replace example rows with your actual non-goals. Each non-goal must have PM/Lead sign-off.
+
+| Non-Goal                                            | Rationale                                                                      | PM/ Lead Agreement |
+|:----------------------------------------------------|:-------------------------------------------------------------------------------|:-------------------|
+| Templating VM spec with the `addresses` annottaion. | It's the responsibility of the 3rd party integrator, and not accessible by us. | [ ] Name/Date      |
+| Creating C/UDN with IP range.                       | Same                                                                           | [ ] Name/Date      |
+
+
+**Important Notes:**
+- Non-goals without stakeholder agreement are considered **risks** and must be escalated (see Section II.7 - Risks and Limitations)
+- Review non-goals during Developer Handoff/QE Kickoff meeting (see Section I.2 - Technology and Design Review)
+
+#### **4. Test Strategy**
+
+##### **A. General**
+Testing this feature should consider and cover the following:
+* see the IP address in the VMI status
+* see it in the guest
+* verify the address is pingable
+* verify this interface can maintain TCP connectivity
+* the feature should be applied on both IPv4/6 families
+* verify the assigned address and connectivity are maintained after VM reboot
```

Comment:
What if you create a VM without the annotation, than stop the VM and add the annotation - after you start the VM it should have the new requested address? I think it worth a test case.